magento/pwa-studio
# [chore] Checkout Page gql updates

## Description

Some minor refactoring of checkout page in preparation for the team.

**Noteable changes:**

1) Using a cart query/cache to check empty state. 
2) A better loading state experience.
3) Simplifying "current step" state to read from a single, locally stored value.
4) Each step/component is now in charge of its "done editing" state so that they can each query for any existing cart data and display accordingly. For now it is represented as `useState`.
5) Moves price adjustments and review button within the payment info component.

## Related Issue





Closes [PWA-395](https://jira.corp.magento.com/browse/PWA-395).

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


1. Add an item and go to checkout page.
2. Refreshing the page should remember the last step you were on. You'll see a weird UX for now because each component is responsible for rendering its "filled/done" state and that is left to later tickets.
3. You should be able to proceed through the steps and "place order".

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
